document/document_test.go

```diff
@@ -510,6 +510,51 @@ func TestNewFromStruct(t *testing.T) {
 		require.NoError(t, err)
 		require.EqualValues(t, 2, v.V.(int64))
 	})
+	t.Run("Embedded struct", func(t *testing.T) {
+		type baz struct {
+			I int `genji:"II"`
+			A string
+		}
+
+		type bar struct {
+			Baz  baz
+			Id   int    `genji:"-"` // ignored
+			Name string `genji:"tag-name"`
+		}
+
+		type foo struct {
+			*bar
+			B   string `genji:"b-b"`
+			Bar bar
+			i   int  // ignored
+			P   *int // should be null
+			A   []byte
+		}
```

Comment:
Instead of declaring new types why don't we use the one that already exists? The `user` struct with the embeded `group`.

Reply:
I prefered wrote a `struct` that I "master"  with my own tests due to the complexity of the the `struct user`.